### PR TITLE
Logprobs suggestions

### PR DESCRIPTION
**Pull request by enochcheung**
## Summary

Some minor re-wording, to clarify some minor things

## Motivation

Why are these changes necessary? How do they improve the cookbook?

I'll comment individually

---

## For new content

When contributing new content, read through our [contribution guidelines](https://github.com/openai/openai-cookbook/blob/main/CONTRIBUTING.md), and mark the following action items as completed:

- [ ] I have added a new entry in [registry.yaml](https://github.com/openai/openai-cookbook/blob/main/registry.yaml) (and, optionally, in [authors.yaml](https://github.com/openai/openai-cookbook/blob/main/authors.yaml)) so that my content renders on the cookbook website.
- [ ] I have conducted a self-review of my content based on the [contribution guidelines](https://github.com/openai/openai-cookbook/blob/main/CONTRIBUTING.md#rubric):
  - [ ] Relevance: This content is related to building with OpenAI technologies and is useful to others.
  - [ ] Uniqueness: I have searched for related examples in the OpenAI Cookbook, and verified that my content offers new insights or unique information compared to existing documentation.
  - [ ] Spelling and Grammar: I have checked for spelling or grammatical mistakes.
  - [ ] Clarity: I have done a final read-through and verified that my submission is well-organized and easy to understand.
  - [ ] Correctness: The information I include is correct and all of my code executes successfully.
  - [ ] Completeness: I have explained everything fully, including all necessary references and citations.

We will rate each of these areas on a scale from 1 to 4, and will only accept contributions that score 3 or higher on all areas. Refer to our [contribution guidelines](https://github.com/openai/openai-cookbook/blob/main/CONTRIBUTING.md) for more details.

### CODE REVIEW

File: examples/Using_logprobs.ipynb [hunk 1]
```diff
@@ -8,11 +8,12 @@
     "\n",
     "This notebook demonstrates the use of the `logprobs` parameter in the Chat Completions API. When `logprobs` is enabled, the API returns the log probabilities of each output token, along with a limited number of the most likely tokens at each token position and their log probabilities. The relevant request parameters are:\n",
     "* `logprobs`: Whether to return log probabilities of the output tokens or not. If true, returns the log probabilities of each output token returned in the content of message. This option is currently not available on the `gpt-4-vision-preview` model.\n",
-    "* `top_logprobs`: An integer between 0 and 5 specifying the number of most likely tokens to return at each token position, each with an associated log probability. logprobs must be set to true if this parameter is used.\n",
+    "* `top_logprobs`: An integer between 0 and 5 specifying the number of most likely tokens to return at each token position, each with an associated log probability. `logprobs` must be set to true if this parameter is used.\n",
```
Comment (enochcheung) [inline examples/Using_logprobs.ipynb:11]: just adding backticks

File: examples/Using_logprobs.ipynb [hunk 2]
```diff
@@ -8,11 +8,12 @@
...
     "\n",
-    "Log probabilities of output tokens indicate the likelihood of each token occurring in the sequence given the context. To simplify, a logprob is `log(p)`, where `p` = probability of a token occurring at a specific position based on the other tokens in the sentence. Some key points about `logprobs`:\n",
-    "* Higher log probabilities suggest a higher likelihood of the token in that context. This allows users to gauge the model's confidence in its output or explore alternative responses the model considered. \n",
-    "* It also allow us to compute the overall probability of a sequence as the sum of the log probs of the individual tokens. This is useful for scoring and ranking model outputs. It's pretty common to take the average logprob of a sentence to choose the best generation.\n",
+    "Log probabilities of output tokens indicate the likelihood of each token occurring in the sequence given the context. To simplify, a logprob is `log(p)`, where `p` = probability of a token occurring at a specific position based on the previous tokens in the context. Some key points about `logprobs`:\n",
```
Comment (enochcheung) [inline examples/Using_logprobs.ipynb:13]: Slightly clarified
- causal masking, distribution only depend on previous tokens
- replace "sentence" with "context"

File: examples/Using_logprobs.ipynb [hunk 3]
```diff
@@ -8,11 +8,12 @@
...
+    "* Higher log probabilities suggest a higher likelihood of the token in that context. This allows users to gauge the model's confidence in its output or explore alternative responses the model considered.\n",
```
Comment (enochcheung) [inline examples/Using_logprobs.ipynb:14]: removed trailing space

File: examples/Using_logprobs.ipynb [hunk 4]
```diff
@@ -8,11 +8,12 @@
...
+    "* Logprob can be any negative number or `0.0`. `0.0` corresponds to 100% probability.\n",
```
Comment (enochcheung) [inline examples/Using_logprobs.ipynb:15]: I think it is sometimes surprising that logprobs are negative. I think it's important to point out, to contextualize the next point to explain why summing results in the joint probablity which should be lower than the individual logprobs

File: examples/Using_logprobs.ipynb [hunk 5]
```diff
@@ -8,11 +8,12 @@
...
+    "* Logprobs allow us to compute the joint probability of a sequence as the sum of the logprobs of the individual tokens. This is useful for scoring and ranking model outputs. Another common approach is to take the average per-token logprob of a sentence to choose the best generation.\n",
```
Comment (enochcheung) [inline examples/Using_logprobs.ipynb:16]: overall -> joint. "overall" can sometimes mean the total probability which is kind of the opposite.

slight re-wording to point out that summing and averaging are two different approaches, where you choose one or the other not both

File: examples/Using_logprobs.ipynb (hunk 4 above)
Comment (shyamal-anadkat) [reply examples/Using_logprobs.ipynb:15]: good point

**Review by shyamal-anadkat — APPROVED** [review]

**Pull request merged**